### ISSUE TITLE
CRITICAL: Notice: Undefined index: dataScopePrefix

### ISSUE THREAD

**Issue by hiteshagrawal84** [EaDesgin/Magento2-City-Dropdown#65]
Hi 

I am getting below error on checkout page

[2019-04-23 07:10:21] report.CRITICAL: Notice: Undefined index: dataScopePrefix in /var/www/html/fatcat/vendor/magento/module-customer-custom-attributes/Block/Checkout/LayoutProcessor.php on line 72 [] []

**Comment by anjuvineesh** [comment]
Is this issue solved yet? I'm also facing the same

**Comment by bayas025** [comment]
Facing same issue, Is there any solution?

**Comment by hiteshgb** [comment]
Yes I have fixed it but I don't remember how i fixed it but if you need code I can provide my code. for city dropdown

**Comment by bayas025** [comment]
Please, It will help me to debug at my end also.

**Comment by hiteshgb** [comment]
Sure no issue. will send you code by tomorrow. just give me a reminder tomorrow.

**Comment by bayas025** [comment]
Sure

**Comment by hiteshgb** [comment]
Also I would like you to join our page https://www.facebook.com/phpdevelopers and spread over your friends, I have created this page to keep updated people about new openings for php and magento developers so needed people can get jobs and also we are posting latest tech news daily basis.

Appreciate if you like to contribute your few secs for us.

**Comment by bayas025** [comment]
could you please give me your code so if it help me to resolve my issue

**Comment by hiteshgb** [comment]
`<?php

namespace Namespace\Checkout\Plugin\Checkout;

use Magento\Checkout\Block\Checkout\LayoutProcessor;
use Namespace\Checkout\Helper\Data;

/**
 * Add area
 */
class LayoutProcessorPlugin
{

    /**
     * @var Data
     */
    protected $helper;

    /**
     * LayoutProcessor constructor.
     * @param Data $helper
     */
    public function __construct(Data $helper)
    {
        $this->helper = $helper;
    }

    /**
     * @param \Magento\Checkout\Block\Checkout\LayoutProcessor $subject
     * @param array $jsLayout
     * @return array
     */
    public function afterProcess(
        LayoutProcessor $subject,
        array  $jsLayout
    ) {
        $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
        ['shippingAddress']['children']['shipping-address-fieldset']['children']['drop_down'] = [
            'component' => 'Ffc_Checkout/js/form/element/city',
            'config' => [
                'customScope' => 'shippingAddress',
                'template' => 'ui/form/field',
                'elementTmpl' => 'ui/form/element/select',
                'id' => 'city',
            ],
            'dataScope' => 'shippingAddress.city',
            'label' => 'Area',
            'provider' => 'checkoutProvider',
            'visible' => true,
            'validation' => [
                "required-entry"    => true,
            ],
            'sortOrder' => 110,
            'id' => 'city',
            'options' => [
                [
                    'value' => '',
                    'label' => 'Please Select',
                ],
            ]

        ];

        if (isset($jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']
            ['children']['shippingAddress']['children']['shipping-address-fieldset']['children'])) {
            $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']
            ['children']['shippingAddress']['children']['shipping-address-fieldset']['children']
            ['telephone'] = $this->helper->telephoneFieldConfig("shippingAddress");
        }

        // Added auto completion address field for google address suggestion
        $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']
        ['children']['shippingAddress']['children']['shipping-address-fieldset']['children']
        ['street']['children'][0]['component'] = 'Amasty_Checkout/js/form/element/autocomplete';


        return $jsLayout;
    }
}//end class
`

**Comment by Elema82** [comment]
Hi! Do I need to solve the same problem @hiteshgb  can you tell me how to add this class in this extension? thanks!

**Comment by hiteshgb** [comment]
@Elema82  You need to add code as per your requirement not complete class as it is as per my store. Please let me know if still you are facing issue.

**Comment by Elema82** [comment]
@hiteshgb  thanks. yes, I founded this error when I tried to enter on the checkout. 
I modified the file 'Eadesigndev\RomCity\Plugin\Checkout\BillingAddressLayoutProcessor.php' on line 50 and I added this line:
 ''dataScopePrefix' => 'shippingAddress' "  

and I still with the same error.

On my store I have magento 2.3.3 version

<img width="794" alt="Captura de Pantalla 2020-03-03 a la(s) 09 30 56" src="https://user-images.githubusercontent.com/25822930/75778669-0368b680-5d37-11ea-96d7-7ff1860ea327.png">
<img width="1336" alt="Captura de Pantalla 2020-03-03 a la(s) 09 58 41" src="https://user-images.githubusercontent.com/25822930/75778676-05cb1080-5d37-11ea-85e5-10ad72ca7ece.png">

<img width="469" alt="Captura de Pantalla 2020-03-03 a la(s) 10 09 18" src="https://user-images.githubusercontent.com/25822930/75778716-18dde080-5d37-11ea-9536-cc43f4d75dc2.png">

**Comment by hiteshgb** [comment]
okay allow me some time I will give you the solution to fix it. please let me know if you are using CE or EE version.

**Comment by Elema82** [comment]
I'm working with magento cloud, I think that is the enterprise edition

**Comment by Elema82** [comment]
hi @hiteshgb I added on this file `Eadesigndev/RomCity/Plugin/Checkout/BillingAddressLayoutProcessor.php` line 87 :

`        $this->result['components']['checkout']['children']['steps']['children']['billing-step']['children']
        ['payment']['children']['payments-list']['children'][$paymentMethodForm]['dataScopePrefix'] = 'customCheckoutForm';`

and I dont see the warning again

**Comment by hiteshagrawal84** [comment]
okay so its working fine now?

**Comment by Elema82** [comment]
this warning? its ok, I found other bugs

**Comment by vrajeshpatel4u** [comment]
Any Update here ? I am also facing same issue.

**Comment by reense** [comment]
@Elema82 did you ever find the root cause of this problem? I'm facing this issue as well on Magento Commerce 2.4.2.

**Comment by giangvdm** [comment]
> hi @hiteshgb I added on this file `Eadesigndev/RomCity/Plugin/Checkout/BillingAddressLayoutProcessor.php` line 87 :
> 
> ` $this->result['components']['checkout']['children']['steps']['children']['billing-step']['children'] ['payment']['children']['payments-list']['children'][$paymentMethodForm]['dataScopePrefix'] = 'customCheckoutForm';`
> 
> and I dont see the warning again

This seems to be a reasonable quick fix for now. It works for me. Thank you!
